### PR TITLE
Code generator

### PR DESCRIPTION
**Pull request by albertoarana**

### CODE REVIEW

File: template/src/components/common/Header.js [hunk 1]
```diff
@@ -56,6 +56,14 @@ export function Header() {
               to='/datasets'
               className={classes.navLink}
             />
+            <Tab
```
Comment (alasarr) [inline template/src/components/common/Header.js:59]: Remove this

**Review by alasarr — CHANGES_REQUESTED** [review]

File: template/src/components/common/Header.js (hunk 1 above)
Comment (albertoarana) [reply template/src/components/common/Header.js:59]: That is menu link question

**Review by alasarr — CHANGES_REQUESTED** [review]
Issues found:

- [ ] Create source with id `Unnamed`
Unnamed is not a correct id.
The name of the view Component should be asked or passed as a parameter.
- `yarn hygen source new dataSourceId` 
-  `yarn hygen view new` The first step is to ask for the source id.

- [ ] Create view Unnamed
Unnamed is not a correct name.
The name of the view Component should be asked or passed as a parameter.
- `yarn hygen view new MyComponent` 
-  `yarn hygen view new` The first step is to ask for the view name.


The title for the Link should be asked after answer `y` to `Do you want a link in the menu?``

- [ ] Create view unassigned class
The compile process complains about this:
```bash
./src/components/views/Unnamed.js
  Line 10:9:  'classes' is assigned a value but never used  no-unused-vars
```

- [ ] Help command

if you execute `yarn hygen help` it says the command failed, but it shouldn't since I've launched a right command

```bash
Error: please specify an action for help.
Hygen v6.0.4

Available actions:
generator: help, new, with-prompt
layer: help, new
source: help, new
view: help, new
error Command failed with exit code 1.
```

**Comment by albertoarana** [comment]
The command `yarn hygen help` has a exit code 1 error by default because you execute nothing. 

The corrects command are: 

* `yarn hygen layer help`
* `yarn hygen source help`
* `yarn hygen view help`

**Review by alasarr — CHANGES_REQUESTED** [review]
Issues:

- [ ] Console errors after create a view

I create a view and the page says:
```javascript
index.js:1 Material-UI: The value provided to the Tabs component is invalid.
None of the Tabs' children match with `country`.
You can provide one of the following values: stores, kpi, datasets, /country.
```

- [ ]`yarn hygen source new` asks for name. it should ask for an id

- [ ] Change the type tileset
It's a new feature 😄 . Instead of Tileset: `BigQuery Tileset`

- [ ] Add dataSource should add the type of the layer
```json
      dataSources: {
      "sqlSource": {
        id: 'test',
        data: 'select * from alberto',
        type: `sql`
      },
      'bigquerySource': {
        id: 'pepe',
        data: 't.a.b',
        type: 'bigquery`
      },
      // Auto import dataSources
    }
```

**Comment by alasarr** [comment]
- [ ] Layer cretion doesn't work

1. Create the source
```bash
$ hygen source new
✔ Name: · testSource
✔ Choose type · SQL dataset
✔ Type a query or the name of your dataset · hola

Loaded templates: _templates
      inject: src/config/cartoSlice.js
✨  Done in 16.59s.
```

2. Crash with the layer
```
$ hygen layer new
✔ Name: · TestLayer
✔ Choose type · CartoSQLLayer
✔ Source id: · testSource
✔ Do you want to attach to some view (y/N) · false
✔ Do you want to attach to some view (y/N) · false

Loaded templates: _templates
ReferenceError: ejs:1
 >> 1| cd <%= cwd %> && yarn hygen view new <%= view %> --type <%= type %> --data <%= data %>

view is not defined
error Command failed with exit code 1.
```

**Comment by albertoarana** [comment]
> Issues:
> 
> * [ ]  Console errors after create a view
> 
> I create a view and the page says:
> 
> ```js
> index.js:1 Material-UI: The value provided to the Tabs component is invalid.
> None of the Tabs' children match with `country`.
> You can provide one of the following values: stores, kpi, datasets, /country.
> ```

This error is an error of boilerplate. When you don't want to add a link, the tabs in Header has a warning.

**Comment by alasarr** [comment]
>This error is an error of boilerplate. When you don't want to add a link, the tabs in Header has a warning.

Not really, it doesn't happen in develop

**Review by alasarr — CHANGES_REQUESTED** [review]
- [x] Asking by ID even if it's in the param
```

yarn hygen source new testSource
yarn run v1.17.3
$ hygen source new testSource
? ID: ›
```

- [x] Error escaping queries adding a Source
If I type a query like this: `select * from alberto where id in ('18')`
 I got:
```
'testSource': {
        id: 'testSource',
        data: 'select * from alberto where id in (&#39;18&#39;)',
        type: 'sql',
      },
```

**Comment by alasarr** [comment]
- [ ] The name of the Layer add a suffix `Layer`
`yarn hygen layer new AlbertoLayer` creates a layer called `AlbertoLayerLayer` it should be `AlbertoLayer`

**Comment by albertoarana** [comment]
> * [ ]  The name of the Layer add a suffix `Layer`
>   `yarn hygen layer new AlbertoLayer` creates a layer called `AlbertoLayerLayer` it should be `AlbertoLayer`

Added automatically Layer, do I remove it?

**Comment by alasarr** [comment]
>Add automatically Layer, do I remove it?
Yes remove it. The user decides the name of the layer, we don't do extra magic

**Comment by alasarr** [comment]
- [ ] Source shouldn't be at cleanup 

When a layer is attached to a view at the cleanup you remove the source . It's not correct, because we don't add the source and if you move to another section when you come back the layer will be fail because the source is not there

**Review by alasarr — APPROVED** [review]

**Pull request merged**